js/src/lib/components/InferenceWidget/shared/WidgetTextarea/WidgetTextarea.svelte

```diff
@@ -5,26 +5,22 @@
 	export let placeholder: string = "Your sentence here...";
 	export let value: string;
 
-	// hack to handle FireFox contenteditable bug
-	let innterHTML: string;
-	let spanEl: HTMLSpanElement;
-	const REGEX_SPAN = /<span .+>(.*)<\/span>/gms;
-	$: {
-		if (spanEl && innterHTML && REGEX_SPAN.test(innterHTML)) {
-			innterHTML = innterHTML.replace(REGEX_SPAN, (_, txt) => {
-				return txt;
-			});
-			spanEl.blur();
-		}
+	function handlePaste(e: ClipboardEvent) {
```

Comment:
It's still buggy: if you copy some text into the area, click outside of it, and select an example, the content is not changed.

Also if you paste with the text content selected, the text is appended at the beginning of the content instead of replacing it.

I feel  this way of doing things is basically breaking the way Svelte handle input / state.

If you're using a span only because textarea don't adjust its height with content (why do we want this anyway ?), I strongly suggest you use a textarea and add some simple code to handle resizing according to its content, it's very easy.

Reply:
(we tried implementing contentEditable element in a React project I work on, and we entered a world of pain doing so - like is seemingly the case here :D )

Reply:
https://github.com/huggingface/huggingface_hub/pull/625/commits/9b34fece7caee5897cf0074f1af326d44c227bb0 wdyt